### PR TITLE
CSV Importer Custom Management Command

### PR DESCRIPTION
**Pull request by PaKZer0**
A small summary of changes:
* Added a Custom Management Command for csv import
* Some test were also added for this new command

### CODE REVIEW

File: src/niweb/apps/scan/migrations/0002_auto_20190121_0856.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (johanlundberg) [inline src/niweb/apps/scan/migrations/0002_auto_20190121_0856.py]: Is this one really part of your changes?

File: src/niweb/apps/noclook/management/commands/csvimport.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+                                parent_org_name,
+                                node_type,
+                                relation_meta_type
+                            )
+
+                        parent_node = parent_organization.get_node()
+                        parent_node.add_property('name', parent_org_name)
+
+            	        # n4: add relation between org and parent_org
+                        graph_node.set_parent(parent_organization.pk)
```
Comment (johanlundberg) [inline src/niweb/apps/noclook/management/commands/csvimport.py]: Not incorrect but I would explicitly use handle_id even though it is the NodeHandles primary key.

File: src/niweb/apps/noclook/management/commands/csvimport.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+                # Print iterations progress
+                if options['verbosity'] > 0:
+                    imported_lines = imported_lines + 1
+                    self.printProgressBar(imported_lines, total_lines)
+
+            csv_contacts.close()
+
+    # replace all the duplicate code
+    def get_or_create(self, node_name, node_type, node_meta_type):
```
[the remark this answers is not in the document]
Comment (johanlundberg) [reply src/niweb/apps/noclook/management/commands/csvimport.py]: There are also some util functions for consumer scripts in https://github.com/SUNET/ni/blob/master/src/scripts/utils.py for getting or creating NodeHandles/Nodes with unique names and or types.

File: src/niweb/apps/noclook/management/commands/csvimport.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                )
+                dbtype.save()
+            else:
+                dbtype = dbtype.first()
+
+        total_lines = 0
+
+        csv_organizations = None
+        csv_contacts = None
+        self.user = User.objects.all().first()
```
[the remark this answers is not in the document]
Comment (PaKZer0) [reply src/niweb/apps/noclook/management/commands/csvimport.py]: I've changed this for:
```
self.user = User.objects.filter(username='noclook').first()
```
what do you think? do we have another neater method for this?

File: src/niweb/apps/noclook/management/commands/csvimport.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                new_organization = self.get_or_create(
+                        account_name,
+                        node_type,
+                        relation_meta_type
+                    )
+
+            	# n4: add attributes
+                graph_node = new_organization.get_node()
+
+                graph_node.add_property('name', account_name)
```
[the remark this answers is not in the document]
Comment (PaKZer0) [reply src/niweb/apps/noclook/management/commands/csvimport.py]: Ok, this felt really silly hehe

File: src/niweb/apps/scan/migrations/0002_auto_20190121_0856.py (hunk 1 above)
Comment (PaKZer0) [reply src/niweb/apps/scan/migrations/0002_auto_20190121_0856.py]: Not really, but this migration was on my changes and I don't really recall the moment this has been generated. I've checked the models and the state of the project and seems to me necessary to commit but let me know if it breaks something. This migration sets a default value for the field, which is what the models.py (from scan I think) says, but the previous migrations doesn't state.

File: src/niweb/apps/noclook/management/commands/csvimport.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (PaKZer0) [reply src/niweb/apps/noclook/management/commands/csvimport.py]: I just didn't know the method existed already, but I changed this anyways for simplicity sake.

File: src/niweb/apps/noclook/management/commands/csvimport.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (johanlundberg) [reply src/niweb/apps/noclook/management/commands/csvimport.py]: You have a util function that is used in other consumer scripts here, https://github.com/SUNET/ni/blob/master/src/scripts/utils.py#L49.

**Pull request merged**